### ISSUE TITLE
Add an automated way to update Explore plugin sources

### ISSUE THREAD

**Issue by dylanmccall** [endlessm/endless-key-flatpak#45]
When there is a new release of [kolibri-explore-plugin](https://github.com/endlessm/kolibri-explore-plugin), we need to manually update [the sources in build-aux/flatpak/modules/python3-kolibri-explore-plugin.json](https://github.com/endlessm/endless-key-flatpak/blob/main/build-aux/flatpak/modules/python3-kolibri-explore-plugin.json#L9-L21) with the new download URLs for that specific release, as well as matching sha256 hashes. This is tedious and annoying.

Instead, let's add a little script which deals with that. The workflow could be something like…

```
./build-aux/scripts/update-explore-plugin-sources.sh [version]
git add build-aux/flatpak/modules/python3-kolibri-explore-plugin.json
git commit
```

Bonus points if this is hooked up in such a way that this dependency is kept up to date automatically.

We should add a similar script for [endless-key-collections](https://github.com/endlessm/endless-key-flatpak/blob/main/build-aux/flatpak/modules/endless-key-collections.json#L8-L15).

**Comment by dbnicholson** [comment]
There's also https://github.com/flathub/flatpak-external-data-checker.

**Comment by dylanmccall** [comment]
Oh yes, I keep forgetting that exists. Yes, use that thing! :)

**Comment by dbnicholson** [comment]
#61 will help keep the dev flatpak up to date, but flathub already does it automatically. What I think we will need is more `x-checker-data` in the both the dev and flathub builds to guide the checker on what to do. See https://github.com/flathub/flatpak-external-data-checker#changes-to-flatpak-manifests.

**Comment by dbnicholson** [comment]
I'm going to make an explore plugin release and see how this reacts.

**Comment by dbnicholson** [comment]
> I'm going to make an explore plugin release and see how this reacts.

Or not. I was planning to backport the Spanish selection to stable, but it's on top of the welcome-screen to loading-screen changes. Well, hopefully we'll be ready to release the next major version next week.

**Comment by erikos** [comment]
Ok, I guess this is another ticket that is dependent on #815 - blocked until then.

I think this is dev-verification.

**Comment by starnight** [comment]
After examine the CI, seems all of the modules are skipped ...
https://github.com/endlessm/endless-key-flatpak/actions/runs/6272283279/job/17033528697

I found @wjt's slide: [Automate your Flatpak manifest updates today!](https://conf.linuxappsummit.org/event/3/contributions/60/attachments/19/52/8%20Will%20Thompson%20%E2%80%93%20Automate%20y%20our%20Flatpak%20manifest%20updates%20today.pdf)  It shows the configuration examples in manifest files.  I think we missed the `x-checker-data` property in modules.

**Comment by dbnicholson** [comment]
I merged #84. I then ran the [update workflow](https://github.com/endlessm/endless-key-flatpak/actions/runs/6330296563). That fell over because f-e-d-c is marking the build-aux/flatpak directory as safe but not actually the workspace root directory. I guess it expects that the manifest is always in the root of the project.

**Comment by dbnicholson** [comment]
> I guess it expects that the manifest is always in the root of the project.

Who was the clown that made that decision? [Whoops](https://github.com/flathub/flatpak-external-data-checker/commit/e55be842) :face_with_spiral_eyes:

**Comment by dbnicholson** [comment]
> > I guess it expects that the manifest is always in the root of the project.
> 
> Who was the clown that made that decision? [Whoops](https://github.com/flathub/flatpak-external-data-checker/commit/e55be842) 😵‍💫

https://github.com/flathub/flatpak-external-data-checker/pull/387

In the meantime, I think we can workaround it by overriding the container entrypoint.

**Comment by dbnicholson** [comment]
[Failed](https://github.com/endlessm/endless-key-flatpak/actions/runs/6331311814/job/17195617164)

**Comment by dbnicholson** [comment]
We have [a](https://github.com/endlessm/endless-key-flatpak/actions/runs/6332786163/job/17199848296) [winner](https://github.com/endlessm/endless-key-flatpak/pull/94)!

**Comment by dbnicholson** [comment]
Now we need to take this over to flathub.

**Comment by dbnicholson** [comment]
The flathub issue was fixed and now it filed an [update PR](https://github.com/flathub/org.endlessos.Key/pull/8). Yay!

**Comment by dbnicholson** [comment]
One last thing we should explore is having the PRs [automatically merged](https://github.com/flathub/flatpak-external-data-checker#automatically-merging-prs). That works both on Flathub as well as in a separate repo.

**Comment by dbnicholson** [comment]
> I presume we can archive this?

Yes, it was able to create the PRs automatically.
